src/MeetingsSDKAdapter.js

```diff
@@ -509,6 +588,181 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
+        // eslint-disable-next-line no-console
+        console.warn(`Stopping local share stream for meeting "${ID}"`);
+        this.stopStream(this.meetings[ID].localShare);
+        this.meetings[ID].localShare = null;
+      }
+
+      sdkMeeting.emit(EVENT_MEDIA_LOCAL_UPDATE, {
+        control: SHARE_CONTROL,
+        state: MeetingControlState.INACTIVE,
+      });
+    };
+
+    //
+    // Workflow:
+    // To enable or to disable the local sharing stream based on toggle state.
+    // Will stop sharing stream and reset UI state when error happens
+    //
+    try {
+      if (this.meetings[ID].localShare) {
+        await disableSharingStream();
+      } else {
+        await enableSharingStream();
+      }
+    } catch (error) {
+      resetSharingStream(error);
+    }
+  }
+
+  /**
+   * Returns an observable that emits the display data of a share control.
+   *
+   * @param {string} ID ID of the meeting to start/stop screen share
+   * @returns {Observable.<MeetingControlDisplay>}
+   * @memberof MeetingJSONAdapter
+   */
+  shareControl(ID) {
+    const sdkMeeting = this.fetchMeeting(ID);
+    const inactiveShare = {
+      ID: SHARE_CONTROL,
+      icon: 'share',
+      tooltip: 'Start Share',
+      state: MeetingControlState.INACTIVE,
+      text: null,
+    };
+    const activeShare = {
+      ID: SHARE_CONTROL,
+      icon: 'share',
+      tooltip: 'Stop Share',
+      state: MeetingControlState.ACTIVE,
+      text: null,
+    };
+    const disabledShare = {
+      ID: SHARE_CONTROL,
+      icon: 'share',
+      tooltip: 'Sharing is Unavailable',
+      state: MeetingControlState.DISABLED,
+      text: null,
+    };
+
+    const getDisplayData$ = Observable.create((observer) => {
+      if (sdkMeeting) {
+        observer.next(inactiveShare);
+      } else {
+        observer.error(new Error(`Could not find meeting with ID "${ID}" to add share control`));
+      }
+
+      observer.complete();
+    });
+
+    const localMediaUpdateEvent$ = fromEvent(sdkMeeting, EVENT_MEDIA_LOCAL_UPDATE).pipe(
+      filter((event) => event.control === SHARE_CONTROL),
+      map(({state}) => {
+        let eventData;
+
+        switch (state) {
+          case MeetingControlState.DISABLED:
+            eventData = disabledShare;
+            break;
+          case MeetingControlState.INACTIVE:
+            eventData = inactiveShare;
+            break;
+          case MeetingControlState.ACTIVE:
+            eventData = activeShare;
+            break;
+          default:
+            eventData = disabledShare;
+            break;
+        }
+
+        return eventData;
+      })
+    );
+
+    const meetingWithMediaStoppedSharingLocalEvent$ = fromEvent(sdkMeeting, EVENT_LOCAL_SHARE_STOP).pipe(
+      // eslint-disable-next-line no-console
+      tap(() => console.log('EVENT_LOCAL_SHARE_STOP is triggered', this)),
+      tap(async () => {
+        try {
+          await this.meetingControls[SHARE_CONTROL].action(ID);
```

Comment:
Why are you invoking the control's action here?

Reply:
sdk will handle the stop event, but it only stops the local share stream, we need to send message to let locus know that the sender now stop the sharing, so we called the action function
I have created the jira here: https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-152004

Reply:
Ok, then `tap` is the wrong operator to use here. We need to use a combination of `flatMap` and `defer` to be able to add the promise as part of the observable stream flow.
```suggestion
          flatMap(defer(() => this.meetingControls[SHARE_CONTROL].action(ID)));
```